src/conv/k-g/sketch.cpp

```diff
@@ -0,0 +1,190 @@
+/*                 S K E T C H . C P P
+ * BRL-CAD
+ *
+ * Copyright (c) 2024 United States Government as represented by
+ * the U.S. Army Research Laboratory.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this file; see the file named COPYING for more
+ * information.
+ */
+ /** @file sketch.cpp
+  *
+  * LS Dyna keyword file to BRL-CAD converter:
+  * intermediate sketch implementation
+  */
+
+#include "sketch.h"
+
+
+
+Sketch::Sketch(void)
+{
+    BU_GET(m_sketch, rt_sketch_internal);
+    m_sketch->magic = RT_SKETCH_INTERNAL_MAGIC;
+    m_sketch->curve.count = 0;
+    m_sketch->vert_count = 0;
+}
+
+void Sketch::setName(const char* value)
+{
+    name = value;
+}
+
+rt_sketch_internal* Sketch::creatSketch
+(
+    std::string sketchName,
+    std::string sectionType,
+    const point_t& node1,
+    const point_t& node2,
+    const point_t& node3,
+    std::vector<double> D
+) {
+    if (sectionType == "SECTION_01") {
+	unsigned long vert_count = 12;
+	unsigned long seg_count  = 12;
+	line_seg* lsg;
+
+	point_t V;
+	VMOVE(V, node1);
+	vect_t r, t, s;
+	vect_t n2n3;
+	r[X] = node2[X] - node1[X];
```

Comment:
VSUB2() macro?

Reply:
I wanted to ask if there are some macros for these kind of operations. Will modify